# Loading SSIS old maps

In [86]:
import pandas as pd
old_maping = pd.read_csv('old_map.csv')

In [87]:
# Creating a new column called new_name.
# else x

old_maping['new_name'] = old_maping['JOB_NAME'].apply(lambda x: x[15:]if x[:15] == 'm_STG_DWH_LOAD_'
                                                      else x[11:] if x[:11] == 'm_DWH_LOAD_'
                                                     else (x[12:] if x[:12] == 'm_GSTG_LOAD_' else x))

In [88]:
# I have add a new column called 'USED_TOOL' so that import the new maping with value USED_TOOL = 'INFORMATICA'
old_maping = old_maping[old_maping.USED_TOOL == 'SSIS']
old_maping.head()

WORK_STRAM                        JOB_NAME JOB_INIT_DATE FREQUENCY  \
0        STG         m_STG_DWH_LOAD_ZDT_ERST     04-JUL-18     Daily   
1        STG         m_STG_DWH_LOAD_ZDT_ERTD     04-JUL-18     Daily   
2        STG   m_STG_DWH_LOAD_ZDT_MCI_ERRLOG     09-MAY-18     Daily   
3        STG     m_STG_DWH_LOAD_ZDT_OBJ_CASE     25-DEC-18     Daily   
4        STG  m_STG_DWH_LOAD_ZDT_SADAD_CHECK     02-DEC-18     Daily   

  USED_TOOL         new_name  
0      SSIS         ZDT_ERST  
1      SSIS         ZDT_ERTD  
2      SSIS   ZDT_MCI_ERRLOG  
3      SSIS     ZDT_OBJ_CASE  
4      SSIS  ZDT_SADAD_CHECK

In [89]:
# We have 503 row in the old mapping.
# rr = old_maping[old_maping['WORK_STRAM'] == 'GSTG']
old_maping.count()
# rr.count()
# rr
old_maping['WORK_STRAM'].unique()

# array(['STG', 'BIDM', 'EXE_DB', 'CRM', 'FAHES', 'PMI', 'RISK', 'TP',
#        'VAT_REPORTS', 'REFUND', 'DWH', 'Config', 'Data Quality', 'GDM',
#        'GSTG'], dtype=object)

# old_maping = old_maping[old_maping['WORK_STRAM'] == ("STG") or old_maping['WORK_STRAM'] == ("DWH")
#                         or old_maping['WORK_STRAM'] == ("GDM")
#                         or old_maping['WORK_STRAM'] == ("GSTG")]
old_maping.count()
# x = old_maping[old_maping['WORK_STRAM'] == ("GSTG")]
# x.count()
old_maping['WORK_STRAM'].unique()


array(['STG', 'BIDM', 'EXE_DB', 'CRM', 'FAHES', 'PMI', 'RISK', 'TP',
       'VAT_REPORTS', 'REFUND', 'DWH', 'Config', 'Data Quality', 'GDM',
       'GSTG'], dtype=object)

# Informatica log Maps

In [115]:
info_log_STG = pd.read_csv('C_INFO_LOGS_V.CSV')
info_log_STG = info_log_STG[info_log_STG['SUBJECT_AREA'] == 'TRM_STG']
# info_log_STG['SUBJECT_AREA'].unique()
print(info_log_STG.count())

info_log_DWH = pd.read_csv('C_INFO_LOGS_V.CSV')
info_log_DWH = info_log_DWH[info_log_DWH['SUBJECT_AREA'] == 'DWH_TRM']
print(info_log_DWH.count())

x = pd.read_csv('C_INFO_LOGS_V.CSV')
print(x.count())

WORKFLOW_NAME       272
OWNER               272
SUBJECT_AREA        272
SESSION_NAME        272
SOURCE_ROW_COUNT    272
TARGET_ROW_COUNT    272
JOB_START_DATE      272
JOB_END_DATE        272
STATUS              272
dtype: int64
WORKFLOW_NAME       178
OWNER               178
SUBJECT_AREA        178
SESSION_NAME        178
SOURCE_ROW_COUNT    178
TARGET_ROW_COUNT    178
JOB_START_DATE      178
JOB_END_DATE        178
STATUS              178
dtype: int64
WORKFLOW_NAME       450
OWNER               450
SUBJECT_AREA        450
SESSION_NAME        450
SOURCE_ROW_COUNT    450
TARGET_ROW_COUNT    450
JOB_START_DATE      450
JOB_END_DATE        450
STATUS              450
dtype: int64


In [91]:
# info_log_STG = pd.read_csv('info_STG_MAP.csv')
# info_log_STG

In [123]:
# I am intersted in teh column SESSION_NAME so I will extract it.
info_log_STG = info_log_STG.iloc[: , [3]]
info_log_STG

SESSION_NAME
0               s_m_STG_TRM_BNKA
1           s_m_STG_TRM_ZDT_VTTD
2             s_m_STG_TRM_BUT000
3         s_m_STG_TRM_ZDT_ZA93_L
4         s_m_STG_TRM_ZDM_BP_ACT
..                           ...
267         s_m_STG_TRM_ZDD_VFBD
268  s_m_STG_TRM_ZDT_SADAD_CHECK
269     s_m_STG_TRM_ZDT_APPL_LOG
270           s_m_STG_TRM_FKKVKP
271           s_m_STG_TRM_BUT050

[272 rows x 1 columns]

In [124]:
# There is 252 STG map on the informatica log.
info_log_STG.count()

SESSION_NAME    272
dtype: int64

In [125]:
# Adding a new column called 'new_name'
info_log_STG['new_name'] = info_log_STG['SESSION_NAME'].apply(lambda x: x[12:] if x[:12] == 's_m_STG_TRM_' 
                                                      else x)

/Users/ahmedzelikha/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [145]:
info_log_STG
len(info_log_STG['new_name'].unique())
# info_log_STG[info_log_STG['SESSION_NAME'].str[:12].unique()]

249

In [127]:
xx = info_log_STG[info_log_STG['SESSION_NAME'].str[:12] != 's_m_STG_TRM_']
xx.head()

SESSION_NAME                  new_name
53  s_m_STG_BP_MAIN_ACTIVITY  s_m_STG_BP_MAIN_ACTIVITY
81      s_m_GenerateParmFile      s_m_GenerateParmFile

# Informatica log for EWD Maps. (To get the EDW Tables)

In [176]:
# info_log_EWH = pd.read_csv('Target.csv')
info_log_DWH.count()
not_need = info_log_DWH[info_log_DWH["SESSION_NAME"].str[:8] !='s_m_DWH_']
not_need

Empty DataFrame
Columns: [SESSION_NAME, new_name]
Index: []

In [119]:
# Adding a new column called 'new_name'
info_log_DWH['new_name'] = info_log_DWH['SESSION_NAME'].apply(lambda x: x[8:] if x[:8] == 's_m_STG_' 
                                                      else x)
info_log_DWH.head()
info_log_DWH.count()

WORKFLOW_NAME       178
OWNER               178
SUBJECT_AREA        178
SESSION_NAME        178
SOURCE_ROW_COUNT    178
TARGET_ROW_COUNT    178
JOB_START_DATE      178
JOB_END_DATE        178
STATUS              178
new_name            178
dtype: int64

In [120]:
# I don't need the GSTG Maps
info_log_DWH = info_log_DWH[info_log_DWH["SESSION_NAME"].str[:8] == 's_m_DWH_']
xxx = info_log_DWH[info_log_DWH["SESSION_NAME"].str[:8] !='s_m_DWH_']
xxx
info_log_DWH.count()

WORKFLOW_NAME       177
OWNER               177
SUBJECT_AREA        177
SESSION_NAME        177
SOURCE_ROW_COUNT    177
TARGET_ROW_COUNT    177
JOB_START_DATE      177
JOB_END_DATE        177
STATUS              177
new_name            177
dtype: int64

In [121]:
# Get the EDW tables names.
info_log_DWH['new_name'] = info_log_DWH['SESSION_NAME'].apply(lambda x: x[4:] if x[:4] == 's_m_' 
                                                      else x)
info_log_DWH

WORKFLOW_NAME    OWNER SUBJECT_AREA                  SESSION_NAME  \
272    wf_DWH_TRM  STG_DWH      DWH_TRM             s_m_DWH_SADAD_LOG   
273    wf_DWH_TRM  STG_DWH      DWH_TRM         s_m_DWH_BP_TYPES_DESC   
274    wf_DWH_TRM  STG_DWH      DWH_TRM         s_m_DWH_BP_ROLES_DESC   
275    wf_DWH_TRM  STG_DWH      DWH_TRM  s_m_DWH_BP_EXT_GOODS_DETAILS   
276    wf_DWH_TRM  STG_DWH      DWH_TRM          s_m_DWH_BP_RELATIONS   
..            ...      ...          ...                           ...   
445    wf_DWH_TRM  STG_DWH      DWH_TRM    s_m_DWH_EXT_RETURN_DETAILS   
446    wf_DWH_TRM  STG_DWH      DWH_TRM     s_m_DWH_PERIOD_DEFINITION   
447    wf_DWH_TRM  STG_DWH      DWH_TRM   s_m_DWH_ICR_PERIOD_DUE_DATE   
448    wf_DWH_TRM  STG_DWH      DWH_TRM   s_m_DWH_RETURN_ITZK_DETAILS   
449    wf_DWH_TRM  STG_DWH      DWH_TRM   s_m_DWH_RETURN_ITZC_DETAILS   

     SOURCE_ROW_COUNT  TARGET_ROW_COUNT  JOB_START_DATE    JOB_END_DATE  \
272            110891            110891  4/3/2020 14:57  4/3/2020 14:58   
273                53                53  4/3/2020 14:57  4/3/2020 14:58   
274               118               118  4/3/2020 14:57  4/3/2020 14:58   
275                 0                 0  4/3/2020 14:58  4/3/2020 14:58   
276              8688              8688  4/3/2020 14:57  4/3/2020 14:58   
..                ...               ...             ...             ...   
445               379               378  4/3/2020 14:58  4/3/2020 14:58   
446              1155              1155  4/3/2020 14:58  4/3/2020 14:58   
447              1357              1357  4/3/2020 14:58  4/3/2020 14:58   
448               101                93  4/3/2020 14:58  4/3/2020 14:58   
449              7280              5677  4/3/2020 14:58  4/3/2020 14:59   

      STATUS                  new_name  
272  Success             DWH_SADAD_LOG  
273  Success         DWH_BP_TYPES_DESC  
274  Success         DWH_BP_ROLES_DESC  
275  Success  DWH_BP_EXT_GOODS_DETAILS  
276  Success          DWH_BP_RELATIONS  
..       ...                       ...  
445  Success    DWH_EXT_RETURN_DETAILS  
446  Success     DWH_PERIOD_DEFINITION  
447  Success   DWH_ICR_PERIOD_DUE_DATE  
448  Success   DWH_RETURN_ITZK_DETAILS  
449  Success   DWH_RETURN_ITZC_DETAILS  

[177 rows x 10 columns]

In [132]:
info_log_DWH = info_log_DWH.iloc[: , [3, 9]]
info_log_DWH

SESSION_NAME                  new_name
272             s_m_DWH_SADAD_LOG             DWH_SADAD_LOG
273         s_m_DWH_BP_TYPES_DESC         DWH_BP_TYPES_DESC
274         s_m_DWH_BP_ROLES_DESC         DWH_BP_ROLES_DESC
275  s_m_DWH_BP_EXT_GOODS_DETAILS  DWH_BP_EXT_GOODS_DETAILS
276          s_m_DWH_BP_RELATIONS          DWH_BP_RELATIONS
..                            ...                       ...
445    s_m_DWH_EXT_RETURN_DETAILS    DWH_EXT_RETURN_DETAILS
446     s_m_DWH_PERIOD_DEFINITION     DWH_PERIOD_DEFINITION
447   s_m_DWH_ICR_PERIOD_DUE_DATE   DWH_ICR_PERIOD_DUE_DATE
448   s_m_DWH_RETURN_ITZK_DETAILS   DWH_RETURN_ITZK_DETAILS
449   s_m_DWH_RETURN_ITZC_DETAILS   DWH_RETURN_ITZC_DETAILS

[177 rows x 2 columns]

# joining the DataFrames (Old_mapping with informatica maps)

In [159]:
# STG
STG = old_maping[old_maping["WORK_STRAM"] == 'STG']
# print(STG.count())
left_STG = pd.merge(STG, info_log_STG, on='new_name', how='left')
left_STG

left_not_match_STG = left_STG[pd.isnull(left_STG["SESSION_NAME"])]
left_match_STG = left_STG[pd.notnull(left_STG["SESSION_NAME"])]
left_not_match_STG.count()
Col_not_match_STG
print (left_STG.count())
print (left_not_match_STG.count())
print (left_match_STG.count())

# left_STG
# left_STG.sort_values(by=['SESSION_NAME'])
# left_STG.to_csv('mapping_resultseeeeeee.csv')
# gk = left_STG.groupby('SESSION_NAME')
# gk.first()
# xxxxx = left_STG.duplicated()
# left_STG=left_STG[left_STG.duplicated()]

WORK_STRAM       250
JOB_NAME         250
JOB_INIT_DATE    250
FREQUENCY        250
USED_TOOL        250
new_name         250
SESSION_NAME     242
dtype: int64
WORK_STRAM       8
JOB_NAME         8
JOB_INIT_DATE    8
FREQUENCY        8
USED_TOOL        8
new_name         8
SESSION_NAME     0
dtype: int64
WORK_STRAM       242
JOB_NAME         242
JOB_INIT_DATE    242
FREQUENCY        242
USED_TOOL        242
new_name         242
SESSION_NAME     242
dtype: int64


In [161]:
# GSTG
GSTG = old_maping[old_maping["WORK_STRAM"] == 'GSTG']
# print(GSTG.count())
left_GSTG = pd.merge(GSTG, info_log_STG, on='new_name', how='left')
left_GSTG

left_not_match_GSTG = left_GSTG[pd.isnull(left_GSTG["SESSION_NAME"])]
left_match_GSTG = left_GSTG[pd.notnull(left_GSTG["SESSION_NAME"])]
left_not_match_GSTG.count()
left_not_match_GSTG
print (left_GSTG.count())
print (left_not_match_GSTG.count())
print (left_match_GSTG.count())
# Col_not_match_GSTG

WORK_STRAM       12
JOB_NAME         12
JOB_INIT_DATE    12
FREQUENCY        12
USED_TOOL        12
new_name         12
SESSION_NAME      0
dtype: int64
WORK_STRAM       12
JOB_NAME         12
JOB_INIT_DATE    12
FREQUENCY        12
USED_TOOL        12
new_name         12
SESSION_NAME      0
dtype: int64
WORK_STRAM       0
JOB_NAME         0
JOB_INIT_DATE    0
FREQUENCY        0
USED_TOOL        0
new_name         0
SESSION_NAME     0
dtype: int64


In [162]:
# DWH
DWH = old_maping[old_maping["WORK_STRAM"] == 'DWH']
# print(DWH.count())
left_DWH = pd.merge(DWH, info_log_DWH, on='new_name', how='left')
left_DWH

left_not_match_DWH = left_DWH[pd.isnull(left_DWH["SESSION_NAME"])]
left_match_DWH = left_DWH[pd.notnull(left_DWH["SESSION_NAME"])]
left_not_match_DWH.count()
left_not_match_DWH
print (left_DWH.count())
print (left_not_match_DWH.count())
print (left_match_DWH.count())
# Col_not_match_DWH

WORK_STRAM       180
JOB_NAME         180
JOB_INIT_DATE    180
FREQUENCY        180
USED_TOOL        180
new_name         180
SESSION_NAME     168
dtype: int64
WORK_STRAM       12
JOB_NAME         12
JOB_INIT_DATE    12
FREQUENCY        12
USED_TOOL        12
new_name         12
SESSION_NAME      0
dtype: int64
WORK_STRAM       168
JOB_NAME         168
JOB_INIT_DATE    168
FREQUENCY        168
USED_TOOL        168
new_name         168
SESSION_NAME     168
dtype: int64


In [163]:
# GDM
GDM = old_maping[old_maping["WORK_STRAM"] == 'GDM']
# print(GDM.count())
left_GDM = pd.merge(GDM, info_log_DWH, on='new_name', how='left')
left_GDM

left_not_match_GDM = left_GDM[pd.isnull(left_GDM["SESSION_NAME"])]
left_match_GDM = left_GDM[pd.notnull(left_GDM["SESSION_NAME"])]
left_not_match_GDM.count()
left_not_match_GDM
print (left_GDM.count())
print (left_not_match_GDM.count())
print (left_match_GDM.count())
# Col_not_match_GDM

WORK_STRAM       13
JOB_NAME         13
JOB_INIT_DATE    13
FREQUENCY        13
USED_TOOL        13
new_name         13
SESSION_NAME      0
dtype: int64
WORK_STRAM       13
JOB_NAME         13
JOB_INIT_DATE    13
FREQUENCY        13
USED_TOOL        13
new_name         13
SESSION_NAME      0
dtype: int64
WORK_STRAM       0
JOB_NAME         0
JOB_INIT_DATE    0
FREQUENCY        0
USED_TOOL        0
new_name         0
SESSION_NAME     0
dtype: int64


# Matched Maps.

In [174]:
# Combine the two dataframes (STG Tables with EDW Tables)

matched_maps = pd.concat([left_match_STG, left_match_DWH])
matched_maps['USED_TOOL'] = 'INFORMATICA'
matched_maps.to_csv('matched_maps.csv', index=False)
com

WORK_STRAM                                 JOB_NAME JOB_INIT_DATE  \
0          STG                  m_STG_DWH_LOAD_ZDT_ERST     04-JUL-18   
1          STG                  m_STG_DWH_LOAD_ZDT_ERTD     04-JUL-18   
2          STG            m_STG_DWH_LOAD_ZDT_MCI_ERRLOG     09-MAY-18   
3          STG              m_STG_DWH_LOAD_ZDT_OBJ_CASE     25-DEC-18   
4          STG           m_STG_DWH_LOAD_ZDT_SADAD_CHECK     02-DEC-18   
..         ...                                      ...           ...   
175        DWH        m_DWH_LOAD_DWH_WH_ADDRESS_DETAILS     04-JUL-18   
176        DWH           m_DWH_LOAD_DWH_WH_BANK_DETAILS     04-JUL-18   
177        DWH          m_DWH_LOAD_DWH_WH_GOODS_DETAILS     04-JUL-18   
178        DWH         m_DWH_LOAD_DWH_WH_HEADER_DETAILS     04-JUL-18   
179        DWH  m_DWH_LOAD_DWH_WH_OWNER_MANGERS_DETAILS     04-JUL-18   

    FREQUENCY    USED_TOOL                      new_name  \
0       Daily  INFORMATICA                      ZDT_ERST   
1       Daily  INFORMATICA                      ZDT_ERTD   
2       Daily  INFORMATICA                ZDT_MCI_ERRLOG   
3       Daily  INFORMATICA                  ZDT_OBJ_CASE   
4       Daily  INFORMATICA               ZDT_SADAD_CHECK   
..        ...          ...                           ...   
175     Daily  INFORMATICA        DWH_WH_ADDRESS_DETAILS   
176     Daily  INFORMATICA           DWH_WH_BANK_DETAILS   
177     Daily  INFORMATICA          DWH_WH_GOODS_DETAILS   
178     Daily  INFORMATICA         DWH_WH_HEADER_DETAILS   
179     Daily  INFORMATICA  DWH_WH_OWNER_MANGERS_DETAILS   

                         SESSION_NAME  
0                s_m_STG_TRM_ZDT_ERST  
1                s_m_STG_TRM_ZDT_ERTD  
2          s_m_STG_TRM_ZDT_MCI_ERRLOG  
3            s_m_STG_TRM_ZDT_OBJ_CASE  
4         s_m_STG_TRM_ZDT_SADAD_CHECK  
..                                ...  
175        s_m_DWH_WH_ADDRESS_DETAILS  
176           s_m_DWH_WH_BANK_DETAILS  
177          s_m_DWH_WH_GOODS_DETAILS  
178         s_m_DWH_WH_HEADER_DETAILS  
179  s_m_DWH_WH_OWNER_MANGERS_DETAILS  

[410 rows x 7 columns]

# Not Matched Maps.

In [172]:

not_matched_maps = pd.concat([left_not_match_DWH, left_not_match_GDM, left_not_match_GSTG, left_not_match_STG])
not_matched_maps.count()
not_matched_maps.to_csv('Not_matched_maps.csv', index=False)
not_matched_maps

WORK_STRAM                                     JOB_NAME JOB_INIT_DATE  \
6          DWH    m_DWH_LOAD_DWH_CMS_REV_VT_SEC_STATUS_DESC     20-FEB-20   
7          DWH                        m_DWH_CUST_PORTS_DESC     03-OCT-18   
83         DWH              m_DWH_LOAD_DWH_CORR_STSTUS_DESC     17-DEC-18   
86         DWH               m_DWH_LOAD_DWH_CUST_CALL_ORDER     16-SEP-18   
88         DWH  m_DWH_LOAD_DWH_CUST_DECLARATION_HEADER_CULG     13-SEP-18   
89         DWH   m_DWH_LOAD_DWH_CUST_DECLARATION_ITEMS_CULG     13-SEP-18   
94         DWH       m_DWH_LOAD_DWH_CUST_OLD_IMPORTS_DETAIL     21-OCT-18   
129        DWH          m_DWH_LOAD_DWH_GOV_CONTRACTS_DETAIL     21-OCT-18   
133        DWH      m_DWH_LOAD_DWH_INSTALLMENT_PLANS_HEADER     30-DEC-18   
134        DWH       m_DWH_LOAD_DWH_INSTALLMENT_PLANS_ITEMS     30-DEC-18   
172        DWH      m_DWH_LOAD_DWH_VT_Amendment_Reason_Desc     04-JUL-18   
173        DWH          m_DWH_LOAD_DWH_VT_DeReg_REASON_DESC     07-OCT-18   
0          GDM                     m_DWH_LOAD_DWH_MOJ_DEALS     05-JAN-19   
1          GDM                     m_DWH_LOAD_DWH_MOJ_PARTY     05-JAN-19   
2          GDM               m_DWH_LOAD_DWH_QWAEM_XBRL_DATA     21-MAR-19   
3          GDM           m_DWH_LOAD_DWH_QWAEM_XBRL_TAG_DESC     21-MAR-19   
4          GDM          m_DWH_LOAD_EXT_ETMAD_ORDERS_PAYMENT     11-SEP-19   
5          GDM                            m_DWH_LOAD_MCI_CR     24-OCT-19   
6          GDM                      m_DWH_LOAD_MCI_CR_ERRLG     24-OCT-19   
7          GDM                          m_DWH_LOAD_MCI_DATA     09-OCT-19   
8          GDM                        m_DWH_LOAD_MCI_ERRLOG     09-OCT-19   
9          GDM                       m_DWH_LOAD_MCI_INQUIRY     20-OCT-19   
10         GDM                       m_DWH_LOAD_QAWAEM_DATA     13-SEP-18   
11         GDM                         m_DWH_LOAD_REG_MOMRA     16-OCT-19   
12         GDM                 m_DWH_LOAD_SAP_INTERFACE_REP     09-OCT-19   
0         GSTG         m_GSTG_LOAD_EXT_ETMAD_ORDERS_PAYMENT     11-SEP-19   
1         GSTG                           m_GSTG_LOAD_MCI_CR     24-OCT-19   
2         GSTG                     m_GSTG_LOAD_MCI_CR_ERRLG     24-OCT-19   
3         GSTG                         m_GSTG_LOAD_MCI_DATA     09-OCT-19   
4         GSTG                       m_GSTG_LOAD_MCI_ERRLOG     09-OCT-19   
5         GSTG                      m_GSTG_LOAD_MCI_INQUIRY     20-OCT-19   
6         GSTG                           m_GSTG_LOAD_QAWAEM     13-SEP-18   
7         GSTG                m_GSTG_LOAD_SAP_INTERFACE_REP     09-OCT-19   
8         GSTG                     m_GSTG_LOAD_ZDD_MOJ_DEAL     05-JAN-19   
9         GSTG                    m_GSTG_LOAD_ZDD_MOJ_PARTY     05-JAN-19   
10        GSTG                    m_GSTG_LOAD_ZDT_XBRL_DATA     21-MAR-19   
11        GSTG                  m_GSTG_LOAD_ZDT_XBRL_TAGMAP     21-MAR-19   
30         STG                     m_STG_DWH_LOAD_FKK_SEC_N     24-FEB-20   
31         STG              m_STG_DWH_LOAD_TFK_SEC_STATUS_T     20-FEB-20   
65         STG               m_STG_DWH_LOAD_DFMCA_FORM_VERS     27-MAY-19   
68         STG          m_STG_DWH_LOAD_EXT_CONTRACTS_DETAIL     21-OCT-18   
98         STG          m_STG_DWH_LOAD_STG_BP_MAIN_ACTIVITY     04-JUL-18   
99         STG      m_STG_DWH_LOAD_STG_TRM_ALL_TRANSACTIONS     22-NOV-18   
100        STG          m_STG_DWH_LOAD_STG_TRM_TRANSACTIONS     22-NOV-18   
165        STG                   m_STG_DWH_LOAD_WEB_IMP_SAP     16-OCT-18   

    FREQUENCY USED_TOOL                          new_name SESSION_NAME  
6       Daily      SSIS    DWH_CMS_REV_VT_SEC_STATUS_DESC          NaN  
7       Daily      SSIS             m_DWH_CUST_PORTS_DESC          NaN  
83      Daily      SSIS              DWH_CORR_STSTUS_DESC          NaN  
86      Daily      SSIS               DWH_CUST_CALL_ORDER          NaN  
88      Daily      SSIS  DWH_CUST_DECLARATION_HEADER_CULG          NaN  
89      Daily      SSIS

# The Final result for mapping 

<B>454</B> table
<B>49</B> datamart
<B>total = 454</B>

STG 241 (249)
<br>DWH 168 (180) </BR>
<br>GDM 0 (13) </BR>
<br>GSTG 0 (12) </BR>

the diffrence is 45 map

In [175]:
# the results.
result = matched_maps.iloc[:, [0, 6, 2, 3, 4]]
result

WORK_STRAM                      SESSION_NAME JOB_INIT_DATE FREQUENCY  \
0          STG              s_m_STG_TRM_ZDT_ERST     04-JUL-18     Daily   
1          STG              s_m_STG_TRM_ZDT_ERTD     04-JUL-18     Daily   
2          STG        s_m_STG_TRM_ZDT_MCI_ERRLOG     09-MAY-18     Daily   
3          STG          s_m_STG_TRM_ZDT_OBJ_CASE     25-DEC-18     Daily   
4          STG       s_m_STG_TRM_ZDT_SADAD_CHECK     02-DEC-18     Daily   
..         ...                               ...           ...       ...   
175        DWH        s_m_DWH_WH_ADDRESS_DETAILS     04-JUL-18     Daily   
176        DWH           s_m_DWH_WH_BANK_DETAILS     04-JUL-18     Daily   
177        DWH          s_m_DWH_WH_GOODS_DETAILS     04-JUL-18     Daily   
178        DWH         s_m_DWH_WH_HEADER_DETAILS     04-JUL-18     Daily   
179        DWH  s_m_DWH_WH_OWNER_MANGERS_DETAILS     04-JUL-18     Daily   

       USED_TOOL  
0    INFORMATICA  
1    INFORMATICA  
2    INFORMATICA  
3    INFORMATICA  
4    INFORMATICA  
..           ...  
175  INFORMATICA  
176  INFORMATICA  
177  INFORMATICA  
178  INFORMATICA  
179  INFORMATICA  

[410 rows x 5 columns]

In [32]:
result.to_csv('mapping_results.csv')